py-rattler/rattler/lock/lock_file.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+import os
+from typing import List, Optional, Tuple
+from rattler.lock.config import (
+    CondaPackageConfig,
+    LockFileChannelConfig,
+    PypiPackageConfig,
+)
+from rattler.lock.environment import Environment
+
+from rattler.rattler import PyLockFile
+
+
+class LockFile:
+    """
+    Represents a lock-file for both Conda packages and Pypi packages.
+    Lock-files can store information for multiple platforms and for multiple environments.
+    """
+
+    _lock_file: PyLockFile
+
+    def __init__(
+        self,
+        channels: Optional[List[LockFileChannelConfig]] = None,
+        conda_packages: Optional[List[CondaPackageConfig]] = None,
+        pypi_packages: Optional[List[PypiPackageConfig]] = None,
```

Comment:
The construction of this object seems illogical to me. I would expect it to take a 

`Dict[str, Environment]`

you can construct a new `Environment` using `Dict[Platform, List[Union[CondaPackage, Tuple[PyPIPackageData, PypiPackageEnvironmentData]]`.

Internally (in rust) you construct a new `Arc<LockFile>` for this data. 

In the constructor of `LockFile` you combine this data again.

Reply:
Removed this constructor for now, will raise a separate PR for it.